Add eprinttype field to General Entry editor tab

prerak09 opened the pull request:
Closes #14555

This pull request adds the `eprinttype` field to the default **General** entry editor tab so that it appears alongside `eprint`, which is where users naturally expect it.  
In addition, a preference migration is included to ensure that existing users who still use the old default General tab configuration are updated automatically, while fully respecting any custom tab setups.

### Steps to test

1. Launch JabRef.
2. Open an existing entry or create a new one.
3. Open the Entry Editor and navigate to the **General** tab.
4. Verify that both `eprint` and `eprinttype` fields are visible in the General tab.
5. i check in my system by running gradle run and checking if `eprint` is added or not .

### Mandatory checks

- [x] I own the copyright of the code submitted and I license it under the MIT license
- [x] I manually tested my changes in running JabRef
- [/] I added JUnit tests for changes (not applicable for preference/UI configuration changes)
- [x] I added screenshots in the PR description (change is visible to the user)
- [x] I described the change in `CHANGELOG.md` in a way that is understandable for the average user
- [/] I checked the user documentation 

<img width="1312" height="626" alt="image" src="https://github.com/user-attachments/assets/8dbca519-5b21-4716-bfec-147c981cf93b" />
<img width="2214" height="1476" alt="image" src="https://github.com/user-attachments/assets/85acc874-ce3c-4f09-bf22-231d43caaed7" />

jabref-machine commented:
You ticked that you modified `CHANGELOG.md`, but no new entry was found there.

If you made changes that are visible to the user, please add a brief description along with the issue number to the `CHANGELOG.md` file. If you did not, please replace the cross (`[x]`) by a slash (`[/]`) to indicate that no `CHANGELOG.md` entry is necessary. More details can be found in our [Developer Documentation about the changelog](https://devdocs.jabref.org/decisions/0007-human-readable-changelog.html).


<!-- ghprcomment
Check PR CHANGELOG.md
CHANGELOG.md needs to be modified if indicated
-->

koppor commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -70,6 +70,7 @@ public static void runMigrations(JabRefGuiPreferences preferences) {
         moveApiKeysToKeyring(preferences);
         removeCommentsFromCustomEditorTabs(preferences);
         addICORERankingFieldToGeneralTab(preferences);
+        addEprintTypeFieldToGeneralTab(preferences);
```
Alphabetic ordering in this "gruop" would be nice - swap these both.

koppor commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -612,6 +613,48 @@ static void addICORERankingFieldToGeneralTab(GuiPreferences preferences) {
      * Thus, the configuration ih the preferences is obsolete
      */
+
+    static void addEprintTypeFieldToGeneralTab(GuiPreferences preferences) {
```
This method is very similar to `addICORERankingFieldToGeneralTab`. They should be unified somehow. - Maybe, `oldDefaultGeneralFields`. This can be a `Set` and if the current field set matches any of that set, then reset to the default.

I think, your `null` check is better than the one of the ICORE method.

koppor reviewed (changes requested):
Good first step. However, there already is a similar method. Please unify. Comment inline.

prerak09 commented:
@koppor i forgot to remove the comments that i usually do to see what new things i added.
please check once and let me know

jabref-machine commented:
Your code currently does not meet JabRef's code guidelines. IntelliJ auto format covers some cases. There seem to be issues with your code style and autoformat configuration. Please reformat your code (<kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>L</kbd>) and commit, then push.

In special cases, consider using `// formatter:off` and `// formatter:on` annotations to allow deviation from the code style.


<!-- ghprcomment
Source Code Tests
format
-->

prerak09 commented:
I unified the migration by extending `addICORERankingFieldToGeneralTab`
The method now checks for the old default General tab field set and resets to the current defaults, which include both ICORE ranking and eprinttype.
This removes duplicated logic and keeps General tab migrations in one place.

koppor commented on CHANGELOG.md:
```diff
@@ -11,6 +11,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 
 ### Added
 
+- Added the `eprinttype` field to the General entry editor tab for users still using the default configuration. (#14555)
```
This is an issue number, but not a proper link.

Please check the line below in the file how to create a proper link.

prerak09 commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -584,21 +584,39 @@ static void moveApiKeysToKeyring(JabRefCliPreferences preferences) {
      * @implNote The default fields for the "General" tab are defined by {@link FieldFactory#getDefaultGeneralFields()}.
      */
     static void addICORERankingFieldToGeneralTab(GuiPreferences preferences) {
```
i am little bit confused what i have to do now
Do I have to change the name of the method or what?
i am a little new i dont understand what to do in what situations will take some time sorry about that

prerak09 commented:
@koppor i changed the method name from icore to migrateGeneralTabDefaultFields now the method migrates the General tab defaults when the user still has an old default setup.
- i also fixed the changelog link problem .
and as instructed i read alpha 3 release now we handle both before alpha 3 and after alha 3 release changes 
for that i split the migration function into two cases
pre and post .

koppor commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -69,7 +70,7 @@ public static void runMigrations(JabRefGuiPreferences preferences) {
         moveApiKeysToKeyring(preferences);
         removeCommentsFromCustomEditorTabs(preferences);
-        addICORERankingFieldToGeneralTab(preferences);
+        migrateGeneralTabDefaultFields(preferences);
```
good

koppor commented on CHANGELOG.md:
```diff
@@ -11,6 +11,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 
 ### Added
 
+- We added the `eprinttype` field to the General entry editor tab. [#14555](https://github.com/JabRef/jabref/issues/14555)
```
Good

koppor commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -583,35 +584,59 @@ static void moveApiKeysToKeyring(JabRefCliPreferences preferences) {
+            return;
+        }
+
+        // Defaults ICORE (before alpha.3)
```
I think, this should be 

```suggestion
        // Defaults fields of tab "General" (before alpha.3)
```

koppor commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -583,35 +584,59 @@ static void moveApiKeysToKeyring(JabRefCliPreferences preferences) {
+        }
+
+        // Defaults ICORE (before alpha.3)
+        Set<Field> preICoreDefaults = Set.of(
```
Maybe even rename this to `prev60alphaFields3` - then you can safe the comment

koppor commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -573,45 +574,68 @@ static void moveApiKeysToKeyring(JabRefCliPreferences preferences) {
+                preferences.getEntryEditorPreferences().getEntryEditorTabs();
+
+        Set<Field> currentGeneralPrefs =
+                entryEditorPrefs.get(Localization.lang("General"));
```
Why line break? - One line plz

prerak09 commented on jabgui/src/main/java/org/jabref/migrations/PreferencesMigrations.java:
```diff
@@ -573,38 +574,50 @@ static void moveApiKeysToKeyring(JabRefCliPreferences preferences) {
      * </p>
      *
-     * @param preferences the user's current preferences
-     * @implNote The default fields for the "General" tab are defined by {@link FieldFactory#getDefaultGeneralFields()}.
```
mistakenly removed 
Sorry
